pydantic_settings/sources.py

```diff
@@ -674,6 +696,309 @@ def __repr__(self) -> str:
+                    if not self.cli_avoid_json:
+                        added_args.append(arg_name)
+                        kwargs['help'] = f'set {arg_name} from JSON string'
+                        model_group.add_argument(f'{arg_flag}{arg_name}', **kwargs)
+                    for model in sub_models:
+                        self._add_fields_to_parser(
+                            parser,
+                            model,
+                            added_args=added_args,
+                            arg_prefix=f'{arg_prefix}{field_name}.',
+                            subcommand_prefix=subcommand_prefix,
+                            group=model_group,
+                        )
+                elif group is not None:
+                    added_args.append(arg_name)
+                    group.add_argument(f'{arg_flag}{arg_name}', **kwargs)
+                else:
+                    added_args.append(arg_name)
+                    parser.add_argument(f'{arg_flag}{arg_name}', **kwargs)
+        return parser
+
+    def _get_modified_args(self, obj: Any) -> tuple[str, ...]:
+        if not self.cli_hide_none_type:
+            return get_args(obj)
+        else:
+            return tuple([type_ for type_ in get_args(obj) if type_ is not type(None)])
+
+    def _metavar_format_list(self, args: list[str]) -> str:
+        if 'JSON' in args:
+            args = args[: args.index('JSON') + 1] + [arg for arg in args[args.index('JSON') + 1 :] if arg != 'JSON']
+        return ','.join(args)
+
+    def _metavar_format_recurse(self, obj: Any) -> str:
+        """Pretty metavar representation of a type. Adapts logic from `pydantic._repr.display_as_type`."""
+        if isinstance(obj, FunctionType):
+            return obj.__name__
+        elif obj is ...:
+            return '...'
+        elif isinstance(obj, (pydantic._internal._repr.Representation, pydantic.v1.utils.Representation)):
+            return repr(obj)
+        elif isinstance(obj, typing_extensions.TypeAliasType):  # type: ignore
+            return str(obj)
+
+        if not isinstance(obj, (typing_base, WithArgsTypes, type)):
+            obj = obj.__class__
+
+        if origin_is_union(get_origin(obj)):
+            args = self._metavar_format_list(list(map(self._metavar_format_recurse, self._get_modified_args(obj))))
+            return f'{{{args}}}' if ',' in args else args
+        elif get_origin(obj) in (typing_extensions.Literal, typing.Literal):
+            args = self._metavar_format_list(list(map(repr, self._get_modified_args(obj))))
+            return f'{{{args}}}' if ',' in args else args
+        elif isinstance(obj, WithArgsTypes):
+            args = self._metavar_format_list(list(map(self._metavar_format_recurse, self._get_modified_args(obj))))
+            return f'{obj.__qualname__}[{args}]'
+        elif obj is type(None):
+            return self.env_parse_none_str
+        elif is_model_class(obj):
+            return 'JSON'
```

Comment:
Any `BaseModel` type in CLI help text will be displayed as `JSON`, is this OK? I felt it is more confusing seeing a class name(s) at the CLI help text as opposed to "JSON". You'll have to refer to the model anyways to know how to structure the JSON input. e.g.

```python
from pydantic import BaseModel
from pydantic_settings import BaseSettings

class SubModel(BaseModel):
    v1: int

class Settings(BaseSettings):
    sub_model: SubModel

# Displaying models as "JSON" 
Settings(_cli_parse_args=['--help'])
"""
usage: example.py [-h] [--sub_model JSON] [--sub_model.v1 int]

options:
  -h, --help          show this help message and exit
sub_model options:
  --sub_model JSON    set sub_model from JSON string
  --sub_model.v1 int 
"""

# Displaying models as class name
Settings(_cli_parse_args=['--help'])
"""
usage: example.py [-h] [--sub_model SubModel] [--sub_model.v1 int]

options:
  -h, --help          show this help message and exit
sub_model options:
  --sub_model SubModel    set sub_model from JSON string
  --sub_model.v1 int 
"""
```

Reply:
I think it is ok